Cabal-syntax/src/Distribution/Compat/Prelude.hs

```diff
@@ -144,6 +143,7 @@ import Prelude                       as BasePrelude hiding
     , Traversable, traverse, sequenceA
     , Foldable, foldMap
     )
+import Data.Foldable as BasePrelude (foldl, elem, sum, product, maximum, minimum)
```

Comment:
Tangent to this set of patches: `maxiumum`, `minimum` are partial too.

```
λ> maximum []
*** Exception: Prelude.maximum: empty list
```

Reply:
Unfortunately, Cabal does use `minimum` / `maximum`, so eliminating them goes beyond the scope of this PR.